### DISCUSSION TITLE
Introducing: chatgpt-demo v2 (now `Anse`)

### DISCUSSION THREAD

**Discussion by ddiu8081** [ddiu8081/chatgpt-demo#247] [Announcements]
### Clear and concise description of the problem

In recent times, we received some very interesting feature requests one after another. Like #63, #87, #90, #231 and so on.

However, the UI we initially provided was designed to be minimalistic and not able to carry so much features.

Therefore, we plan to update our major UI to provide more panels for more features.

Here's our rough sketch:  

![Index](https://user-images.githubusercontent.com/1998168/224493079-28590472-f9d3-4db6-8285-c531261f4b99.png)

#### Our Figma file 

https://www.figma.com/file/LupJ77idVjxFHq4Ox54KiG/Chatgpt-demo-v2

### Development

We started a new repo and started work on the new UI. 

**Repo**: [anse-app/anse](https://github.com/anse-app/anse)
**Preview**: https://anse.app

### Contribution

Feel free to make your further suggestions in this issue, it will help us a lot.

**Comment by jerryokk** [comment]
#228 Hope to have voice input function.

**Reply by AdamEbid1** [reply]
This is indeed poggers

**Comment by 1592363624** [comment]
nice

**Comment by dingqiangit** [comment]
cool! 希望UI可以支持中文或者多语言的自适应和切换

**Reply by AdamEbid1** [reply]
Very Cool!

**Reply by Zhou-Shilin** [reply]
i18n

**Comment by Hime-Hina** [comment]
重新生成按钮会放在MessageItem外面吗？目前要是fetch failed就得刷新页面，用户消息还在历史记录里面，但是处于没有调用api的状态，就不得不删除记录重新发送或者再发一次。

**Comment by Axx-03** [comment]
希望能添加针对之前对话的编辑功能，对话数据有法子缓存嘛？

**Comment by lxxxv5** [comment]
新UI会兼容手机端显示吗

**Comment by SkyNetX007** [comment]
1. 希望可以将对话导入/导出为JSON或者单纯导出聊天记录为markdown/图片/PDF
~~2. 感觉可以改名为 chatgpt-web 了~~

**Comment by Crabor** [comment]
UI可以学习下platform.openai.com/playground

**Comment by Zoranner** [comment]
建议可以融入任务场景的概念，比如通过自定义System Role模版或者Prompt模版，创建一个任务场景配置，在这个配置下可以实现某些特定任务场景的问答。

**Reply by all666666all** [reply]
主要是Prompt模版

**Comment by Limour-dev** [comment]
希望对某个对话可以加一个分享功能，生成一个链接https://domain/hash，将相应的设置和对话记录保存到服务器，在其他地方访问该链接可以将该对话的设置和对话记录恢复到本地，方便分享给小伙伴或迁移到其他设备。

**Comment by Aa435161** [comment]
改好如果問題的問題，回答很長的時候
有辦法在第一段回答結束後下一段回答可以繼續下去

**Comment by Limour-dev** [comment]
建议增加对多个key的支持：
```js
const apiKey = import.meta.env.OPENAI_API_KEY.split("|")
const get1Key = (arr) => {
    const randomIndex = Math.floor(Math.random() * arr.length);
    const randomSubstr = arr[randomIndex];
    return randomSubstr;
}
const initOptions = generatePayload(get1Key(apiKey), messages)
```

**Comment by Lmmmmmm-bb** [comment]
Would like to ask when v2 be released.

**Comment by vaynevayne** [comment]
我觉得没必要开多个chat, 就用现在的就挺好的, 搞复杂了倒难用,多开几个浏览器tab不就好了,保持简单吧 哥哥

**Comment by Zoranner** [comment]


**Reply by vaynevayne** [reply]
我觉得丑爆了 好吗

**Reply by erbanku** [reply]
> 原评论已删除，你们说的都对。 你们开心就好，溜了溜了......
@Zoranner 

开玩笑，我觉得挺好的，继续加油吧，不要太在意别人的想法😊

**Comment by AashishNandakumar** [comment]
Nice job with the ui

**Comment by snekkenull** [comment]
looks awesome!

**Comment by fg0578fg0578** [comment]
Great job!
The new version is not released yet?

**Comment by birdmanmandbir** [comment]
i can't edit chat title on mobile phone,seems it have not been test on mobile platform

**Comment by birdmanmandbir** [comment]
Please add export history function, i also wonder how does history stores

**Reply by ddiu8081** [reply]
Data import and export are not available for now. 

All data are saved in the browser's indexedDB. You can open devTools to view the storage structure.

**Comment by meesii** [comment]
![image](https://user-images.githubusercontent.com/1815046/233769041-a755f190-ebe0-4cb8-aa96-50e6c4c4b85b.png)
utools里面的一个免费插件，UI设计的很不错，可以借鉴一下。

**Comment by yiGmMk** [comment]
In v2 branch,I have to set OPAN_AI_KEY in the page ("API Key") even if environment variable OPAN_AI_KEY was set in netlify.I also checked it has value after deploy. bug?

**Reply by ddiu8081** [reply]
Yes, v2 does not read environment variables yet. This is in our roadmap and we will add this soon.

**Reply by yiGmMk** [reply]
got it

**Comment by marisarbq** [comment]
Thank you for your creation. Is it possible to add the function of importing and exporting conversation data based on your current capabilities in the future, to facilitate sharing and storage?

**Comment by zzy-life** [comment]
https://github.com/Chanzhaoyu/chatgpt-web
Their ui is nice, but they don't support Netlify, so I gave up on it.

**Comment by carlostipton1319** [comment]
We received a series of interesting feature requests (#63, #87, #90, #231, etc.), but our current minimalistic UI design is unable to accommodate them all. As a result, we are planning a [Top AI](https://topaiblog.com/) major UI update to incorporate additional panels for these features.

**Comment by saxill** [comment]
Great work,
I'm building my own **Ai assistance** hence i request to approach you on behalf on this, asking help for this

**Comment by saxill** [comment]
Great work,
I'm building my own **Ai assistance** hence i request to approach you on behalf on this, asking help for this
